src/memray/reporters/flamegraph.py

```diff
@@ -117,87 +261,78 @@ def _from_any_snapshot(
                 size = record.size
                 n_allocations = record.n_allocations
 
-            if size is not None:
-                root["value"] += size
-            if n_allocations is not None:
-                root["n_allocations"] += n_allocations
-
-            current_frame_id = 0
-            current_frame = root
+            record_data: RecordData = {
+                "thread_name": thread_id,
+                "size": size,
+                "n_allocations": n_allocations,
+                "intervals": intervals,
+            }
 
             stack = (
                 tuple(record.hybrid_stack_trace())
                 if native_traces
                 else record.stack_trace()
             )
-            num_skipped_frames = 0
-            is_import_system = False
-            for index, stack_frame in enumerate(reversed(stack)):
-                node_key = (current_frame_id, stack_frame, thread_id)
-                if node_key not in node_index_by_key:
-                    if is_cpython_internal(stack_frame):
-                        num_skipped_frames += 1
-                        continue
-
-                    if not is_import_system:
-                        is_import_system = is_frame_from_import_system(stack_frame)
-
-                    new_node_id = len(frames)
-                    node_index_by_key[node_key] = new_node_id
-                    current_frame["children"].append(new_node_id)
-                    frames.append(
-                        create_framegraph_node_from_stack_frame(
-                            stack_frame,
-                            import_system=is_import_system,
-                            thread_id=thread_id,
+
+            # normal flamegraph
+            if not inverted:
+                stack_it = enumerate(reversed(stack))
+                cls.generate_frames(
+                    stack_it=stack_it,
```

Comment:
I'm gonna drop this local variable as well, and just construct it in the call to the function below, since it's only used in one spot, and to keep limiting local state.